### PR TITLE
Card input

### PR DESCRIPTION
**Pull request by DoroVernier**
Könnte gemerged werden.

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+  "stories": [
+    "../stories/**/*.stories.mdx",
+    "../stories/**/*.stories.@(js|jsx|ts|tsx)"
```
Comment (NikRoe) [inline .storybook/main.js:4]:
```suggestion
    "../components/**/*.stories.@(js|jsx|ts|tsx)"
```

File: components/GlobalStyle.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { createGlobalStyle } from "styled-components";
+
+const GlobalStyle = createGlobalStyle`
+    *, *::after, *::before {
+        box-sizing: border-box
+       
+    }
+    body {
+        list-style: none; 
+        background-color: #eee;
```
Comment (NikRoe) [inline components/GlobalStyle.js:10]: 🤩

File: components/card.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (NikRoe) [inline components/card.js]: the file should also start with a capital letter

File: components/GlobalStyle.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+       
+    }
+    body {
+        list-style: none; 
+        background-color: #eee; 
+        margin: 0; 
+        padding: 0; 
+        border: 5px solid black; 
+        width: 376px; 
+        height: 667px;
```
Comment (NikRoe) [inline components/GlobalStyle.js]: do you need this?

File: components/card.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import AddCard from "./input";
+
+
+export default function Card({title, author,isbn}) {
+//  const [newBook, setNewBook] = useState(""); 
+
+// function  addBook(savedBook) {
+//     setNewBook([savedBook, ...newBook]); 
+
+// }
```
Comment (NikRoe) [inline components/card.js]: when you create a pull request, you should remove all comments beforehand

File: components/card.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        //     <AddCard onAddBook={addBook} />
+        //      {newBook.map((book) => {
+        //         return (
+        //             <span key={book.id}>
+        //                 <bookWish/>
+        //             </span>
+        //         ); 
+        //     })} 
+        //     </Container>
+        // </Booklist>
```
Comment (NikRoe) [inline components/card.js]: as above

File: components/input.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import {nanoid} from 'nanoid';
+
+export default function AddCard(onAddBook) {
```
Comment (NikRoe) [inline components/input.js]:
```suggestion
export default function AddCard({onAddBook}) {
```

File: components/input.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  function handleSubmit(event) {
+    event.preventDefault(); 
+
+    const form = event.target; 
+    const bookWish= form.bookWish.value.trim(); 
+
+    const NewCard = {id:nanoid(), bookWish: bookWish}; 
+
+onAddBook(NewCard.value);
```
Comment (NikRoe) [inline components/input.js]:
```suggestion
onAddBook(NewCard); 
```

File: components/input.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+onAddBook(NewCard.value); 
+  form.reset(); 
+
+  }  
+   
+    return (
+        
+<form onSubmit={handleSubmit}>
+    <label htmlFor="bookWish">Find your book </label>
+    <input type="text" name="bookWish" id="nanoid()"  placeholder="Name, Titel or ISBN" maxlength="70"required />
```
Comment (NikRoe) [inline components/input.js]:
```suggestion
    <input type="text" name="bookWish" id="bookWish"  placeholder="Name, Titel or ISBN" maxlength="70"required />
```

File: components/input.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import {nanoid} from 'nanoid';
+
```
Comment (NikRoe) [inline components/input.js]: the file should have the same name as the component `AddCard.js` in this case

File: components/card.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
+
+
+import styled from "styled-components"; 
+import AddCard from "./input";
```
Comment (NikRoe) [inline components/card.js]: we won't need this import :)

File: pages/index.js [hunk 12]
```diff
@@ -1,69 +1,35 @@
...
-        <h1 className={styles.title}>
-          Welcome to <a href="https://nextjs.org">Next.js!</a>
-        </h1>
 
-        <p className={styles.description}>
-          Get started by editing{' '}
-          <code className={styles.code}>pages/index.js</code>
-        </p>
+export default function WishList() {
+ const [newBook, setNewBook] = useState("");
```
Comment (NikRoe) [inline pages/index.js]:
```suggestion
 const [bookList, setBookList] = useState([]); 
```

File: pages/index.js [hunk 13]
```diff
@@ -1,69 +1,35 @@
...
+export default function WishList() {
+ const [newBook, setNewBook] = useState(""); 
 
-        <div className={styles.grid}>
-          <a href="https://nextjs.org/docs" className={styles.card}>
-            <h2>Documentation &rarr;</h2>
-            <p>Find in-depth information about Next.js features and API.</p>
-          </a>
+function  addBook(savedBook) {
+    setNewBook([savedBook, ...newBook]);
```
Comment (NikRoe) [inline pages/index.js]:
```suggestion
    setBookList([savedBook, ...bookList]); 
```

File: pages/index.js [hunk 14]
```diff
@@ -1,69 +1,35 @@
...
-            <Image src="/vercel.svg" alt="Vercel Logo" width={72} height={16} />
-          </span>
-        </a>
-      </footer>
+    <div>
+      <Card title="Grist" author="Abra Berens" isbn=""/>
+      <Card title="The wolf and the woodsman" author="Ava Reid" isbn=""/>
+      <Card title="Dark Earth" author="Rebecca Stott" isbn=""/>
+      <AddCard onAddBook={addBook} />
+           {newBook.map((book) => {
```
Comment (NikRoe) [inline pages/index.js]:
```suggestion
           {bookList.map((book) => {
```

File: pages/index.js [hunk 15]
```diff
@@ -1,69 +1,35 @@
...
+                 return (
+                     <span key={book.id}>
+                         <bookWish/>
```
Comment (NikRoe) [inline pages/index.js]:
```suggestion
                         <book.bookWish/>
```

File: stories/Button.jsx [hunk 16]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
```
Comment (NikRoe) [inline stories/Button.jsx]: you can delete the whole folder this file is in :)

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hey Doro,

this looks like a promising start :) 

I made a few remarks you might want to consider :)

File: components/card.js (hunk 3 above)
Comment (DoroVernier) [reply components/card.js]: changed.

File: components/card.js (hunk 11 above)
Comment (DoroVernier) [reply components/card.js]: removed

File: components/card.js (hunk 5 above)
Comment (DoroVernier) [reply components/card.js]: cleaned up

File: components/card.js (hunk 6 above)
Comment (DoroVernier) [reply components/card.js]: as above

File: components/input.js (hunk 10 above)
Comment (DoroVernier) [reply components/input.js]: is now CreateCard.js

File: pages/index.js (hunk 12 above)
Comment (DoroVernier) [reply pages/index.js]: changed accordingly

File: pages/index.js (hunk 13 above)
Comment (DoroVernier) [reply pages/index.js]: adjusted

File: pages/index.js (hunk 14 above)
Comment (DoroVernier) [reply pages/index.js]: changed

File: components/CreateCard.js [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+export default function CreateCard({ onAddBook }) {
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const form = event.target;
+    const title = form.title.value;
+    const author = form.author.value;
+
+    const newCard = { id: nanoid(), title: title, author: author };
+    console.log(newCard);
```
Comment (K-Juno) [inline components/CreateCard.js:13]: I think you should remove the console.log. :)

**Pull request merged**